source/dub/dub.d

```diff
@@ -1319,6 +1330,7 @@ class Dub {
 		settings.compiler = getCompiler(compiler_binary); // TODO: not using --compiler ???
 		settings.platform = settings.compiler.determinePlatform(settings.buildSettings, compiler_binary, m_defaultArchitecture);
 		settings.buildType = "debug";
+		if (m_defaultLowMemory) settings.buildSettings.options |= BuildOption.lowmem;
 		settings.run = true;
```

Comment:
the "run tool" functions should really be refactored, but that won't be part of this PR, so I'm fine with this change.